### PR TITLE
ramips: Add new device zbt-link we2802d support.

### PR DESCRIPTION
**Pull request by axxop**
MT7628DAN (580MHz), 16MB SPI NOR, 64MB DDR2 RAM.

All works great.

Signed-off-by: likai <likaisoftware@gmail.com>

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
Hi, some formal issues:
- No capitalization after "ramips:" in the commit title
- No dot at the end. I would also remove the word "support" there
- Add a little more data about the device and add flashing instructions to the commit message
- Use a full real name for your From: and Signed-off-by:

File: target/linux/ramips/dts/mt7628an_zbtlink_zbt-we2802d.dts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		led-upgrade = &led_wlan;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_zbtlink_zbt-we2802d.dts]: Try to set compatible to interrupt-driven "gpio-keys" and remove poll-interval.

File: target/linux/ramips/image/mt76x8.mk [hunk 2]
```diff
@@ -573,6 +573,14 @@ define Device/zbtlink_zbt-we1226
 endef
 TARGET_DEVICES += zbtlink_zbt-we1226
 
+define Device/zbtlink_zbt-we2802d
+  MTK_SOC := mt7628an
+  IMAGE_SIZE := 16064k
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk]: Your firmware partition is 7872k?!

File: target/linux/ramips/dts/mt7628an_zbtlink_zbt-we2802d.dts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	};
+};
+
+&wmac {
+	status = "okay";
+	ralink,mtd-eeprom = <&factory 0x4>;
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x2e>;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_zbtlink_zbt-we2802d.dts]: Is there another MAC address in 0x28?
Can you check which of the 0x28/0x2e address correspond to LAN and WAN addresses on stock firmware?
Have you checked that WiFi gets the same MAC address as on stock firmware?

**Comment by psyborg55** [comment]
This can not be accepted. Read general guidelines on adding new device support

**Comment by 981213** [comment]
@psyborg55 

Again... Saying "it can not be accepted" without providing a reason isn't a helpful review at all. A proper response would be what @adrianschmutzler have already done and since he has done it you don't need to repeat it again.

**Comment by psyborg55** [comment]
> without providing a reason

are you fucking blind or what?

**Comment by adrianschmutzler** [comment]
@psyborg55 

Your behavior is unacceptable. Despite the OpenWrt "Be nice to each other", your post would obviously violate the normal rules of discussion in forums or when discussing in person.

Unfortunately, there is no option to report your most recent post, so please just delete it yourself.

**Comment by adrianschmutzler** [comment]
@axxop Please do not feel repelled by the response in this particular pull request.
I have been reviewing lots of PRs during the last months and your case is really the first where I have seen such negative initial feedback.

Consider yourself invited to reopen the request when you have addressed the requested changes. I hope you do understand that OpenWrt has quality standards that need to be met before a PR can be merged.

**Comment by psyborg55** [comment]
having a committer that doesn't read or understand behind first sentence, or another one like you that defends first one immediately while he's obviously wrong is unacceptable.
sounds almost like some gross guy bitching from another PR that was closed twice here already and he was asked not to open any more PRs but he still opened 3rd one.......

**Comment by 981213** [comment]
> are you fucking blind or what?

Well... You didn't actually provide a reason did you?

> Read general guidelines on adding new device support

This is not a "reason" at all.

**Comment by 981213** [comment]
I'll just do some cleanups here.

**Comment by psyborg55** [comment]
yeah, clean your ignorance

**Pull request closed without merging**